test/js/client.js

```diff
@@ -0,0 +1,129 @@
+
+describe("client", function() {
+
+  afterEach(function() {
+    nock.cleanAll();
+  });
+
+  describe("constructor", function() {
+    it("fails if not given address or discovery", function() {
+      assert.throws(function() {
+        new Client();
+      }, "Cannot initialize swagger-test without discovery or address");
+    });
+
+    it("initializes given `address`", function() {
+      c = new Client({address: mockAddress});
+      assert.equal(c.address, mockAddress);
+    });
+
+    describe("initializing given `discovery`", function() {
+      beforeEach(function() {
+        delete process.env.SERVICE_SWAGGER_TEST_DEFAULT_PROTO;
+        delete process.env.SERVICE_SWAGGER_TEST_DEFAULT_HOST;
+        delete process.env.SERVICE_SWAGGER_TEST_DEFAULT_PORT;
+        delete process.env.SERVICE_SWAGGER_TEST_DEFAULT_PROTO;
+        delete process.env.SERVICE_SWAGGER_TEST_DEFAULT_HOST;
+        delete process.env.SERVICE_SWAGGER_TEST_DEFAULT_PORT;
+      });
+
+      it("fails with no env vars", function() {
+        assert.throws(function() {
+          new Client({discovery: true});
+        }, "Missing env var SERVICE_SWAGGER_TEST_DEFAULT_PROTO");
+      });
+
+      it("works with default expose", function() {
+        process.env.SERVICE_SWAGGER_TEST_DEFAULT_PROTO = "http"
+        process.env.SERVICE_SWAGGER_TEST_DEFAULT_HOST = "localhost"
+        process.env.SERVICE_SWAGGER_TEST_DEFAULT_PORT = "8000"
+        c = new Client({discovery: true});
+        assert.equal(c.address, mockAddress);
+      });
+
+      it("works with legacy `http` expose", function() {
+        process.env.SERVICE_SWAGGER_TEST_HTTP_PROTO = "http"
+        process.env.SERVICE_SWAGGER_TEST_HTTP_HOST = "localhost"
+        process.env.SERVICE_SWAGGER_TEST_HTTP_PORT = "8000"
+        c = new Client({discovery: true});
+        assert.equal(c.address, mockAddress);
+      });
+    });
+  });
+
+  describe("GET", function() {
+    it("correctly fills out path, header and query params", function(done) {
+      const c = new Client({address: mockAddress});
+      const params = {
+        authorization: "Bearer YOGI",
+        bookID: "the-mediocre-gatsby",
+        randomBytes: "4" // Very random
+      }
+      const mockResponse = {hello: "world"};
+      const scope = nock(mockAddress, {reqheaders: {'authorization': params.authorization}})
+        .get(`/v1/books/${params.bookID}`)
+        .query({randomBytes: params.randomBytes})
+        .reply(200, mockResponse);
+      c.getBookByID(params, function(err, resp) {
+        assert(scope.isDone());
+        assert.ifError(err);
+        assert.deepEqual(resp, mockResponse)
+        done();
+      });
+    });
+
+    it("accepts an options argument", function(done) {
+      const c = new Client({address: mockAddress});
+      const scope = nock(mockAddress)
+        .get(`/v1/books`)
+        .reply(200);
+      c.getBooks({}, {}, function() {
+        assert(scope.isDone());
+        done();
+      });
+    });
```

Comment:
this test feels almost like it's testing nothing... maybe leave it to the specific option tests, e.g. the retry tests testing the retry policy option